src/datasources/accounts/accounts.datasource.spec.ts

```diff
@@ -62,65 +63,71 @@ describe('AccountsDatasource tests', () => {
 
   describe('createAccount', () => {
     it('creates an account successfully', async () => {
-      const address = getAddress(faker.finance.ethereumAddress());
+      const createAccountDto = createAccountDtoBuilder().build();
 
       const result = await target.createAccount({
-        address,
+        createAccountDto,
         clientIp: faker.internet.ipv4(),
       });
 
       expect(result).toStrictEqual({
         id: expect.any(Number),
         group_id: null,
-        address,
+        address: createAccountDto.address,
+        name: createAccountDto.name,
         created_at: expect.any(Date),
         updated_at: expect.any(Date),
       });
 
       // check the account is stored in the cache
-      const cacheDir = new CacheDir(`account_${address}`, '');
+      const cacheDir = new CacheDir(`account_${createAccountDto.address}`, '');
       const cacheContent = await fakeCacheService.hGet(cacheDir);
       expect(JSON.parse(cacheContent as string)).toStrictEqual(
         expect.arrayContaining([
           expect.objectContaining({
             id: expect.any(Number),
             group_id: null,
-            address,
+            address: createAccountDto.address,
+            name: createAccountDto.name,
           }),
         ]),
       );
     });
 
     it('creates an account successfully if the clientIp is not a valid IP', async () => {
-      const address = getAddress(faker.finance.ethereumAddress());
+      const createAccountDto = createAccountDtoBuilder().build();
 
       const result = await target.createAccount({
-        address,
+        createAccountDto,
         clientIp: faker.string.sample(),
       });
 
       expect(result).toStrictEqual({
         id: expect.any(Number),
         group_id: null,
-        address,
+        address: createAccountDto.address,
+        name: createAccountDto.name,
         created_at: expect.any(Date),
         updated_at: expect.any(Date),
       });
 
       // check the account is stored in the cache
-      const cacheDir = new CacheDir(`account_${address}`, '');
+      const cacheDir = new CacheDir(`account_${createAccountDto.address}`, '');
       const cacheContent = await fakeCacheService.hGet(cacheDir);
       expect(JSON.parse(cacheContent as string)).toStrictEqual(
         expect.arrayContaining([
           expect.objectContaining({
             id: expect.any(Number),
             group_id: null,
-            address,
+            address: createAccountDto.address,
+            name: createAccountDto.name,
           }),
         ]),
       );
     });
 
+    it.todo('should fail if the name already exists');
```

Comment:
👀

Reply:
😅  Added in [2069930](https://github.com/safe-global/safe-client-gateway/pull/2081/commits/2069930ce18518a38b763ad69b1a7585e6f2016e)